shop/src/test/java/ro/msg/learning/shop/integration/OrderCreationIntegrationTest.java

```diff
@@ -0,0 +1,104 @@
+package ro.msg.learning.shop.integration;
+
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.test.context.junit4.SpringRunner;
+import ro.msg.learning.shop.ShopApplication;
+import ro.msg.learning.shop.controller.TestController;
+import ro.msg.learning.shop.entity.Customer;
+import ro.msg.learning.shop.entity.OrderDetail;
+import ro.msg.learning.shop.entity.Orders;
+import ro.msg.learning.shop.exception.OutOfStockException;
+import ro.msg.learning.shop.service.CustomerService;
+import ro.msg.learning.shop.service.OrderService;
+
+import java.time.LocalDateTime;
+import java.util.Arrays;
+import java.util.List;
+
+@RunWith(SpringRunner.class)
+//@SpringBootTest
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.MOCK,
+        classes = ShopApplication.class)
+@ActiveProfiles("test")
+@TestPropertySource(locations = "classpath:application-integrationtest.properties")
+
+public class OrderCreationIntegrationTest extends TestBase {
+    @Autowired
+    private CustomerService customerService;
+    @Autowired
+    private OrderService orderService;
+    @Autowired
+    TestController testController;
```

Comment:
Do not use directly the Controller here, but perform the requests using MockMvc (see example in the training materials)